### DISCUSSION TITLE
Date / Datepicker widget

### DISCUSSION THREAD

**Discussion by yofisim** [algolia/instantsearch#4959] [Ideas] [answered]
Il would be great to finally have a date widget to filter by timestamp;

Use cases:
- Search within dates (Airbnb, hotels, flights etc...)
- Search prior / further a specific date.

Create my own widget is a bit of a headache;

**Comment by Haroenv** [comment]
Thanks for the request, at the moment we have indeed not yet made a date widget as the use case is very broad, however we have a an example here on how to integrate a third-party UI for the date picker.

https://github.com/algolia/instantsearch.js/tree/master/examples/calendar-widget

**Comment by yofisim** [comment, marked as answer]
Thanks for sharing the use case! Funny I made almost the same code :)

const ONE_DAY_IN_MS = 3600 * 24 * 1000;
const datePicker = instantsearch.connectors.connectRange(
(renderOptions, isFirstRender) => {
if (!isFirstRender) return;
const { refine } = renderOptions;
new Calendar({
element: $('#calendar'),
current_date: Date.now(),
same_day_range: true,
days_array: ['Dim', 'Lun', 'Mar', 'Mer', 'Jeu', 'Ven', 'Sam'],
presets: [{
label: 'Hier',
start: moment().subtract(1, 'day').startOf('day'),
end: moment().subtract(1, 'day').endOf('day')
}, {
label: 'Aujourd\'hui',
start: moment().startOf('day'),
end: moment().endOf('day')
}, {
label: 'Cette Semaine',
start: moment().startOf('week'),
end: moment().endOf('week')
}, {
label: 'Ce Mois',
start: moment().startOf('month'),
end: moment().endOf('month')
}, {
label: 'Cette année',
start: moment().startOf('year'),
end: moment().endOf('year')
}],
callback: function () {
let startYear = moment(this.start_date).year();
let startMonth = moment(this.start_date).month();
let startDay = moment(this.start_date).date();

let endYear = moment(this.end_date).year();
let endMonth = moment(this.end_date).month();
let endDay = moment(this.end_date).date();

let start = new Date(Date.UTC(startYear, startMonth, startDay, 0, 0, 0, 0)).
getTime();
let end = new Date(Date.UTC(endYear, endMonth, endDay, 23, 59, 59, 999)).
getTime();
const actualEnd = start === end ? end + ONE_DAY_IN_MS - 1 : end;
refine([start, actualEnd]);
},
});
}
);
